### PR TITLE
Make instance_of work with tuple like isinstance and unittest's asser…

### PR DESCRIPTION
**Pull request by mouadino**
…tIsInstance

### CODE REVIEW

**Comment by mouadino** [comment]
FYI tests are failing using tox in OSX so relaying on travis :)

File: tests/hamcrest_unit_test/core/isinstanceof_test.py [hunk 1]
```diff
@@ -23,6 +23,8 @@ class IsInstanceOfTest(MatcherTest):
     def testEvaluatesToTrueIfArgumentIsInstanceOfASpecificClass(self):
         self.assert_matches('same class', instance_of(int), 1)
 
+        self.assert_matches('same class', instance_of((int, long)), 1)
```
Comment (offbyone) [inline tests/hamcrest_unit_test/core/isinstanceof_test.py]: I'd like to see something that tests with the second argument; maybe `instance_of((int, datetime), datetime.now())`

File: src/hamcrest/core/core/isinstanceof.py [hunk 2]
```diff
@@ -11,21 +11,26 @@ class IsInstanceOf(BaseMatcher):
...
         self.expected_type = expected_type
 
     def _matches(self, item):
         return isinstance(item, self.expected_type)
 
     def describe_to(self, description):
+        try:
+            type_description = self.expected_type.__name__
+        except AttributeError:
+            type_description = str(self.expected_type)
```
Comment (offbyone) [inline src/hamcrest/core/core/isinstanceof.py]: I'd rather see a better description here, possibly `"one of %s" % ",".join(str(e) for e in self.expected_type)`

**Comment by offbyone** [comment]
If you want this to turn up in 1.8, you'll want to rebase it against the release-1.8 branch. Otherwise it'll roll into 2.0 when I get around to releasing it.

File: tests/hamcrest_unit_test/core/isinstanceof_test.py (hunk 1 above)
Comment (mouadino) [inline tests/hamcrest_unit_test/core/isinstanceof_test.py]: will do.

**Comment by mouadino** [comment]
@offbyone thanks for the review, I will see what I can do for the rebasing part, PR updated BTW.

**Pull request closed without merging**